exec/java-exec/src/main/java/org/apache/drill/exec/compile/JaninoClassCompiler.java

```diff
@@ -58,5 +56,27 @@ public JaninoClassCompiler(ClassLoader parentClassLoader, boolean debug) {
   }
 
   @Override
+  public Map<String,byte[]> compile(final ClassNames className, final String sourceCode)
+      throws CompileException, IOException, ClassNotFoundException {
+
+    ClassFile[] classFiles = doCompile( className, sourceCode );
+    Map<String,byte[]> results = new HashMap<>( );
+    for(int i = 0;  i < classFiles.length;  i++) {
+      ClassFile classFile = classFiles[i];
+      results.put( classFile.getThisClassName(), classFile.toByteArray() );
+    }
+    return results;
+  }
+
+  public ClassFile[] doCompile(final ClassNames className, final String sourceCode)
```

Comment:
1. className parameter is never used.
2. May be make method private as in JDKClassCompiler?